content/200-concepts/200-database-connectors/08-cockroachdb.mdx

```diff
@@ -82,3 +82,67 @@ The following table lists any other current known limitations of CockroachDB com
 | Searching on `String` fields without a full text index not supported                             | Client        | Searching on `String` fields without a full text index (e.g. `where: { text: { search: "cat & dog", }, },`) is currently [not supported in CockroachDB](https://github.com/cockroachdb/cockroach/issues/7821)                                          |
 | Integer division not supported                                                                   | Client        | Integer division (e.g. `data: { int: { divide: 10, }, }`) is currently [not supported in CockroachDB](https://github.com/cockroachdb/cockroach/issues/41448)                                                                                           |
 | Limited filtering on `Json` fields                                                               | Client        | Currently CockroachDB [only supports](https://github.com/cockroachdb/cockroach/issues/49144) `equals` and `not` filtering on `Json` fields                                                                                                             |
+
+## Type mapping between CockroachDB and the Prisma schema
+
+The CockroachDB connector maps the [scalar types](/concepts/components/prisma-schema/data-model#scalar-fields) from the Prisma [data model](/concepts/components/prisma-schema/data-model) as follows to native column types:
+
+> Alternatively, see the [Prisma schema reference](/reference/api-reference/prisma-schema-reference#model-field-scalar-types) for type mappings organized by Prisma type.
+
+### Mapping from Prisma to CockroachDB types on migration
+
+| Prisma     | CockroachDB      |
+| ---------- | ---------------- |
+| `String`   | `STRING`         |
+| `Boolean`  | `BOOL`           |
+| `Int`      | `INT4`           |
+| `BigInt`   | `INT8`           |
+| `Float`    | `FLOAT8`         |
+| `Decimal`  | `DECIMAL(65,30)` |
+| `DateTime` | `TIMESTAMP(3)`   |
+| `Json`     | `JSONB`          |
+| `Bytes`    | `BYTES`          |
+
+### Mapping from CockroachDB to Prisma types on Introspection
+
+When introspecting a CockroachDB database, the database types are mapped to Prisma according to the following table:
+
+| CockroachDB (Type \| Aliases)                | Prisma     | Supported | Native database type attribute       | Notes |
+| -------------------------------------------- | ---------- | :-------: | :----------------------------------- | :---- |
+| `INT` \| `BIGINT`, `INTEGER`                 | `BigInt`   |    ✔️     | `@db.Int8`                           |       |
+| `BOOL` \| `BOOLEAN`                          | `Bool`     |    ✔️     | `@db.Boolean`\*                      |       |
```

Comment:
In the [Postgres connector](https://www.prisma.io/docs/concepts/database-connectors/postgresql#native-type-mappings) several rows have a 'Default mapping for  <type> - no type attribute added to schema' note - are these still relevant for Cockroach?

Reply:
Yes, evfor every distinct value in the `Prisma` column, one of them is the default and is usually not rendered. (So `BigInt` instead of `BigInt @db.Int8` as it is the only one, for the others Tom will be able to provide the values.)

How we display that is pretty bad though indeed. Suggestion: Maybe a column "Default mapping, not rendered in schema" that has a checkbox for all the default ones instead of the repeating message in the Note?

Reply:
Yes, we should revisit this for all connectors at the same time for consistency. Let's merge this one :+1: